src/components/VioletEnroll/VioletEnrollModal.tsx

```diff
@@ -0,0 +1,79 @@
+import { useWeb3React } from '@web3-react/core'
+import useDebounce from 'hooks/useDebounce'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { useAppDispatch } from 'state/hooks'
+import { useIsUserRegisteredWithViolet } from 'state/registration/hooks'
+import { updateRegistrationState } from 'state/registration/reducer'
+import { useIsRegisteredWithViolet } from 'utils/temporary/useIsRegistered'
+
+import Modal from '../Modal'
+import VioletEnroll from './VioletEnroll'
+
+export default function VioletEnrollModal() {
+  const { account, isActive } = useWeb3React()
+  const walletFinishedLoading = useDebounce(isActive, 600)
+  const dispatch = useAppDispatch()
+  const { isRegistered } = useIsRegisteredWithViolet({ ethereumAddress: account })
+  const alreadyRegistered = useIsUserRegisteredWithViolet()
```

Comment:
What's the difference between these?

Reply:
Ok after seeing the rest of the code I understand the distinction of user enrolled vs mauve seeing this wallet before
Let's rename the latter, it is very confusing with the other one